logstash-core/spec/logstash/util/wrapped_synchronous_queue_spec.rb

```diff
@@ -80,12 +80,14 @@
             write_client.push_batch(batch)
 
             read_batch = read_client.read_batch.to_java
-            sleep(0.1) # simulate some work for the `duration_in_millis`
-            # TODO: this interaction should be cleaned in an upcoming PR,
-            # This is what the current pipeline does.
-            read_client.add_filtered_metrics(read_batch.filteredSize)
-            read_client.add_output_metrics(read_batch.filteredSize)
-            read_client.close_batch(read_batch)
+            read_client.execute_with_timers do
```

Comment:
this is pretty awful. We are effectively mimicking the specific chain of events that the `WorkerLoop` does, so this is liable to drift. Migrating these _out_ of the read client is deferred scope.